javalib/src/main/resources/scala-native/uuid_getentropy.c

```diff
@@ -0,0 +1,77 @@
+#include <errno.h>
+
+#ifdef __APPLE__
+#include <sys/random.h>
+#endif
+
+#ifndef _WIN32
+#include <unistd.h>
+#else // _WIN32
+// https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/
+//     realloc?view=msvc-170
+//     rand-s?view=msvc-170
+#define _CRT_RAND_S
+#include <stdlib.h>
+#include <string.h>
+#endif // _WIN32
+
+/* This file is not a candidate for either 'posixlib' or 'clib'.
+ * 'getentropy()' and 'rand_s1' are not defined in either POSIX, nor IEEE/ISO.
+ */
+
+int scalanative_uuid_getentropy(void *buffer, size_t length, int *error) {
+
+#if defined(__linux__) || defined(__FreeBSD__) || defined(__APPLE__)
+    /* Linux:
+     * URL https://man7.org/linux/man-pages/man3/getentropy.3.html
+     * content dated: 2021-03-22
+     * Other BSDs should work but have not been tested.
+     *
+     * On macOS there is no real need for the overhead of "arc4random()"
+     * because only a small number of bytes are being fetched,
+     */
+
+    int status = getentropy(buffer, length);
```

Comment:
At least for BSD/macOS (?) it says this:
> However, it should be noted that [getentropy](https://keith.github.io/xcode-man-pages/getentropy.2.html#getentropy~3)() is primarily intended for use in the construction and seeding of userspace PRNGs like arc4random(3) or CC_crypto(3).

https://keith.github.io/xcode-man-pages/getentropy.2.html

Thoughts?

Reply:
Ah, hold onto your hat, for here we go down a rabbit hole.  This is one of the discussions I had been hoping to 
avoid until the overall 'use OS entropy' had support.  How much lipstick do we want to put on this
`Sus domesticus`?

I spent hours on  `arc4random`, probably more than on the Windows implementation.
I searched the Web and considered design alternatives.  I have learned over time
to trust the FreeBSD documentation and do exactly the __opposite__ of the macOS
documentation.  

Linux, which is not macOS, used to have a similar description, but
its CSPRNG got good enough that the current recommendation is
to rely on `getentropy()`, well, to be truthful, the `getrandom()` which
the former calls.

There is a comment in the code:
```
* On macOS there is no real need for the overhead of "arc4random()"        
* because only a small number of bytes are being fetched,
```
`arc4random` appears to have its own groups of intense supporters
and equally intense opponents.
`arc4random` actually uses ChaCha20 rather than arc4 in its
implementations of the past few years.  It is my understanding
that is slated for even more change, perhaps to maintain
parity with Linux. I believe the `use arc4random` suggestion is
or is about to become obsolete.  

In the case where UUIDs are being generated at a slow rate,
say one or two per second, `getentropy` should have
minimal effect on the system entropy pool.  

Feeding the results of `getentropy` to `arc4random` and spinning
up a new random number generator at each and every call adds
a _lot_ of overhead and is, IMO, more likely to impact the system.

In Industrial strength situations, where billions of UUIDs are being
generated, a durable object is needed which initializes `arc4random`
once (and reseeds occasionally on a timely basis) this would look
a lot like `SecureRandom` but have the drawbacks of a private
implementation. Such situations would be using RNG hardware
instead of `fake SecureRandom`. 

I have no problem is someone wants to implement such an
Industrial solution. I believe that is is far beyond the needs
of `UUID.randomUUID`.  That method had zero internal uses.
Now it is used only by the `UUIDTest.scala` of this PR.

Arman, does your code use `UUID.randomUUID`?  Is that how
you discovered the security flaw?   If yes, how many randomUUIDs 
per second do you require?